packages/flutter/test/widgets/reorderable_list_test.dart

```diff
@@ -0,0 +1,198 @@
+
+    void check({ List<int> visible = const <int>[], List<int> hidden = const <int>[] }) {
+      for (final int i in visible) {
+        expect(find.text('item $i'), findsOneWidget);
+      }
+      for (final int i in hidden) {
+        expect(find.text('item $i'), findsNothing);
+      }
+    }
+
+    // The SliverReorderableList is 800x600, 8 items, each item is 800x100 with
+    // an "item $index" text widget at the item's origin.  Drags are initiated by
+    // a simple press on the text widget.
+    await tester.pumpWidget(TestList(items: items));
+    check(visible: <int>[0, 1, 2, 3, 4, 5], hidden: <int>[6, 7]);
+
+    // Drag item 0 downwards less than halfway and let it snap back. List
+    // should remain as it is.
+    await pressDragRelease(const Offset(12, 50), const Offset(12, 60));
+    check(visible: <int>[0, 1, 2, 3, 4, 5], hidden: <int>[6, 7]);
+    expect(tester.getTopLeft(find.text('item 0')), Offset.zero);
+    expect(tester.getTopLeft(find.text('item 1')), const Offset(0, 100));
+    expect(items, orderedEquals(<int>[0, 1, 2, 3, 4, 5, 6, 7]));
+
+    // Drag item 0 downwards more than halfway to displace item 1.
+    await pressDragRelease(tester.getCenter(find.text('item 0')), const Offset(0, 151));
+    check(visible: <int>[0, 1, 2, 3, 4, 5], hidden: <int>[6, 7]);
+    expect(tester.getTopLeft(find.text('item 1')), Offset.zero);
+    expect(tester.getTopLeft(find.text('item 0')), const Offset(0, 100));
+    expect(items, orderedEquals(<int>[1, 0, 2, 3, 4, 5, 6, 7]));
+
+    // Drag item 0 back to where it was.
+    await pressDragRelease(tester.getCenter(find.text('item 0')), const Offset(0, -51));
+    check(visible: <int>[0, 1, 2, 3, 4, 5], hidden: <int>[6, 7]);
+    expect(tester.getTopLeft(find.text('item 0')), Offset.zero);
+    expect(tester.getTopLeft(find.text('item 1')), const Offset(0, 100));
+    expect(items, orderedEquals(<int>[0, 1, 2, 3, 4, 5, 6, 7]));
+
+    // Drag item 1 to item 3
+    await pressDragRelease(tester.getCenter(find.text('item 1')), const Offset(0, 251));
+    check(visible: <int>[0, 1, 2, 3, 4, 5], hidden: <int>[6, 7]);
+    expect(tester.getTopLeft(find.text('item 0')), Offset.zero);
+    expect(tester.getTopLeft(find.text('item 1')), const Offset(0, 300));
+    expect(tester.getTopLeft(find.text('item 3')), const Offset(0, 200));
+    expect(items, orderedEquals(<int>[0, 2, 3, 1, 4, 5, 6, 7]));
+
+    // Drag item 1 back to where it was
+    await pressDragRelease(tester.getCenter(find.text('item 1')), const Offset(0, -200));
+    check(visible: <int>[0, 1, 2, 3, 4, 5], hidden: <int>[6, 7]);
+    expect(tester.getTopLeft(find.text('item 0')), Offset.zero);
+    expect(tester.getTopLeft(find.text('item 1')), const Offset(0, 100));
+    expect(tester.getTopLeft(find.text('item 3')), const Offset(0, 300));
+    expect(items, orderedEquals(<int>[0, 1, 2, 3, 4, 5, 6, 7]));
+  });
+
+  testWidgets('ReorderableList, items inherit DefaultTextStyle, IconTheme', (WidgetTester tester) async {
```

Comment:
Not sure, but I think it's:
```suggestion
  testWidgets('SliverReorderableList, items inherit DefaultTextStyle, IconTheme', (WidgetTester tester) async {
```